backend/src/main/java/gov/cdc/usds/simplereport/api/apiuser/UserAccountCreationController.java

```diff
@@ -2,49 +2,77 @@
 
 import static gov.cdc.usds.simplereport.config.WebConfiguration.USER_ACCOUNT_REQUEST;
 
+import gov.cdc.usds.simplereport.api.model.errors.InvalidActivationLinkException;
+import gov.cdc.usds.simplereport.api.model.useraccountcreation.SetRecoveryQuestionRequest;
+import gov.cdc.usds.simplereport.api.model.useraccountcreation.UserAccountCreationRequest;
+import gov.cdc.usds.simplereport.idp.authentication.OktaAuthentication;
 import javax.annotation.PostConstruct;
-import javax.servlet.http.HttpSession;
+import javax.servlet.http.HttpServletRequest;
+import org.json.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 /** Controller used for user account creation. */
-// NOTE: This class is not currently functional; it's a WIP so that the frontend has endpoints to
-// query.
 @RestController
 @RequestMapping(USER_ACCOUNT_REQUEST)
 public class UserAccountCreationController {
   private static final Logger LOG = LoggerFactory.getLogger(UserAccountCreationController.class);
 
+  private static final String STATE_TOKEN_KEY = "stateToken";
+  private static final String USER_ID_KEY = "userId";
+
+  @Autowired private OktaAuthentication _oktaAuth;
+
   @PostConstruct
   private void init() {
     LOG.info(
         "WIP: User account request creation REST endpoint enabled. Not for use in production at this time.");
   }
 
   /**
-   * WIP Validates that the requesting user has been sent an invitation to SimpleReport, ensures the
-   * given password meets all requirements, and sets the password in Okta. If the password doesn't
-   * meet requirements, sends a notice back to the frontend.
+   * Validates that the requesting user has been sent an invitation to SimpleReport, ensures the
+   * given password meets all requirements, and sets the password in Okta.
    *
-   * @param session
-   * @return the session id (temporary)
+   * @param requestBody contains the password
+   * @param request contains all header information, including the activation token.
+   * @throws InvalidActivationLinkException if the activation token is invalid.
+   * @throws OktaAuthenticationFailureException if the password is invalid or if the user is not in
+   *     a RESET_PASSWORD state.
    */
   @PostMapping("/initialize-and-set-password")
-  public String setPassword(HttpSession session) {
-    return session.getId();
+  public void activateAccountAndSetPassword(
+      @RequestBody UserAccountCreationRequest requestBody, HttpServletRequest request)
+      throws Exception {
+    JSONObject oktaResponse =
+        _oktaAuth.activateUser(
+            requestBody.getActivationToken(),
+            request.getHeader("X-Forwarded-For"),
```

Comment:
This thread seems resolved, but I did want to chime in to say that we **do** expect multiple addresses in the `X-Forwarded-For` header. Requests that hit the backend should have passed through Akamai, Azure App Gateway, and the App Service load balancer and have a header that looks like `X-Forwarded-For: <client IP>, <Akamai IP>, <App Gateway IP>` with the App Service LB as the direct peer IP.

In theory, you should be appending the IP of your direct peer to the `X-Forwarded-For` list so that it comes out like `X-Forwarded-For: <client IP>, <Akamai IP>, <App Gateway IP>, <App Service LB IP>`, but I'm not sure if Okta cares about anything other than the originating client's IP.